### PR TITLE
Improved the readability of the debugging statistics

### PR DESCRIPTION
**Pull request by weegeekps**
Changed the UI of the debugging statistics a bit to improve readability:

- Real values are delimited by spaces.
- We have a human readable form of the "Tileset Cached Bytes" now.

image

### CODE REVIEW

File: exts/cesium.omniverse/cesium/omniverse/ui/statistics_widget.py [hunk 1]
```diff
@@ -20,14 +25,35 @@ class CesiumOmniverseStatisticsWidget(ui.Frame):
...
+                StatisticsItem(label=TILESET_CACHED_BYTES_TEXT, value=self._tileset_cached_bytes_model),
+                StatisticsItem(
+                    label=TILESET_CACHED_BYTES_HUMAN_READABLE_TEXT,
+                    value=self._tileset_cached_bytes_human_readable_model,
+                ),
+            ]
+        )
+        self._statistics_delegate = StatisticsDelegate()
+        self._statistics_tree_view: Optional[ui.TreeView] = None
+        self._aframe: Optional[ui.Frame] = None
```
Comment (lilleyse) [inline exts/cesium.omniverse/cesium/omniverse/ui/statistics_widget.py:56]: `_aframe` doesn't appear to be used

File: exts/cesium.omniverse/cesium/omniverse/ui/statistics_widget.py [hunk 2]
```diff
@@ -53,33 +79,27 @@ def _on_update_frame(self, _e: carb.events.IEvent):
...
-            TILESET_CACHED_BYTES_TEXT.format(render_statistics.tileset_cached_bytes)
-        )
+        self._materials_loaded_model.set_value(render_statistics.number_of_materials_loaded)
+        self._geometries_loaded_model.set_value(render_statistics.number_of_geometries_loaded)
+        self._geometries_visible_model.set_value(render_statistics.number_of_geometries_visible)
+        self._triangles_loaded_model.set_value(render_statistics.number_of_triangles_loaded)
+        self._triangles_visible_model.set_value(render_statistics.number_of_triangles_visible)
+        self._tileset_cached_bytes_model.set_value(render_statistics.tileset_cached_bytes)
+        self._tileset_cached_bytes_human_readable_model.set_value(render_statistics.tileset_cached_bytes)
+        self._statistics.refresh()
```
Comment (lilleyse) [inline exts/cesium.omniverse/cesium/omniverse/ui/statistics_widget.py:89]: I'm seeing a drop from 82 FPS to 73 FPS just by having the Cesium Debugging open even after that stray log is removed. This doesn't seem to happen on main. I'm guessing it's related to this line and items being marked as changed every frame.

File: exts/cesium.omniverse/cesium/omniverse/ui/models/human_readable_bytes_model.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+    def get_value_as_int(self) -> int:
+        return self._value
+
+    def get_value_as_float(self) -> float:
+        return float(self._value)
+
+    def get_value_as_string(self) -> str:
+        for unit in ["B", "KiB", "MiB", "GiB", "TiB", "PiB", "EiB", "ZiB"]:
+            logging.getLogger(__name__).warn(self._value)
```
[the remark this answers is not in the document]
Comment (weegeekps) [reply exts/cesium.omniverse/cesium/omniverse/ui/models/human_readable_bytes_model.py:29]: Yup. Fixed.

File: exts/cesium.omniverse/cesium/omniverse/ui/models/human_readable_bytes_model.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+            if abs(self._value) < 1024:
+                return f"{self._value:3.2f} {unit}"
+            self._value /= 1024
```
[the remark this answers is not in the document]
Comment (weegeekps) [reply exts/cesium.omniverse/cesium/omniverse/ui/models/human_readable_bytes_model.py:32]: Good catch. Fixed.

File: exts/cesium.omniverse/cesium/omniverse/ui/statistics_widget.py (hunk 1 above)
Comment (weegeekps) [reply exts/cesium.omniverse/cesium/omniverse/ui/statistics_widget.py:56]: Doh. Removed.

**Review by corybarr — APPROVED** [review]

File: exts/cesium.omniverse/cesium/omniverse/ui/statistics_widget.py (hunk 2 above)
Comment (weegeekps) [reply exts/cesium.omniverse/cesium/omniverse/ui/statistics_widget.py:88]: Oh wow, yikes.

I don't think it's related to the refresh. I just commented out all of my usages of `SpaceDelimitedNumberModel` and performance was stable. I actually think it has to do with the fact we're using that f-string stuff. Strings in Python are notoriously slow so it makes sense too. I will rethink my approach for those view models on Monday.

**Comment by weegeekps** [comment]
@lilleyse Try out my latest changes. That should improve performance.

**Comment by lilleyse** [comment]
Yup, performance is better now. Thanks @weegeekps.

**Pull request merged**